src/macaron/slsa_analyzer/checks/build_as_code_check.py

```diff
@@ -127,6 +127,21 @@ def _has_deploy_command(self, commands: list[list[str]], build_tool: BaseBuildTo
                         return str(com)
         return ""
 
+    def _add_to_result_table(self, facts: BuildAsCodeFacts, check_result: CheckResult) -> None:
+        """Add to the check result result tables, creating the list if it doesn't yet exist.
+
+        Parameters
+        ----------
+        facts : BuildAsCodeFacts
+            Facts to add to the result tables.
+        check_result : CheckResult
+            The object containing result data of a check.
+        """
+        if "result_tables" in check_result.keys():
```

Comment:
If I understand correctly, the `result_tables` field in the `CheckResult` instance passed to the check has already been initialized to an empty list (see [here](https://github.com/oracle/macaron/blob/386c085d9ad30dd9eb9b35aef0c19b91718e02f7/src/macaron/slsa_analyzer/checks/base_check.py#L81)). 
Then wouldn't we need to check again to see if `result_tables` is defined. So I guess you could just append new `BuildAsCodeFacts` instances to it directly. @behnaz Please let me know if my understanding is correct.

Reply:
I did try just .append, but there somehow were scenarios where I'd be getting a `KeyError` for `result_tables`, so I thought it would be best to just do it like this to be safe. If its an issue though I can try digging into why that happened

Reply:
Managed to reproduce the KeyError, I missed though that it was happening in `test_travis_ci_deploy` and not the code itself. Latest commit should fix that + reverts to just `.append()`s